apps/omg/test/support/conformance.ex

```diff
@@ -0,0 +1,125 @@
+    # elixir implementation errors
+    assert {:error, _} = Transaction.decode(some_binary)
+
+    # solidity implementation errors
+    some_binary
+    |> solidity_hash(contract)
+    |> assert_contract_reverted()
+
+    true
+  end
+
+  @doc """
+  Check if both implementations either:
+    - treat distinct transactions as distinct but produce sign hashes consistently
+    - both error
+  _under the condition that `tx2_binary` decodes fine in the "native" implementation in Elixir_
+  """
+  def verify_distinct_or_erroring(tx1_binary, tx2_binary, contract) do
+    # TODO - think of a better approach to handling the different treatment of valid/admissible tx/output types
+    #      there shouldn't be that many cases, 2 (`{:ok, _}` and `{:error, _}`) should ideally do
+    case Transaction.decode(tx2_binary) do
+      # if the mutated transaction decodes fine, we check whether signature hashes match across impls and are distinct
+      {:ok, _} ->
+        verify_distinct(Transaction.decode!(tx1_binary), Transaction.decode!(tx2_binary), contract)
+
+      # NOTE: unrecognized tx/output type is never picked up in the contract, since there, decoding assumes already a
+      #       particular type (i.e. Payment) and only checks if delivered type (`1`, `2`, ...) is correct in later stage
+      #       when fetching and verifying the `ISpendingCondition`
+      {:error, :unrecognized_transaction_type} ->
+        true
+
+      {:error, :unrecognized_output_type} ->
+        true
+
+      # NOTE: another temporary special case handling, until a better idea comes. `tx_type` 3 is `FeeTokenClaim`
+      #       transaction which pops out as `malformed` in `elixir-omg` and is accepted by contracts
+      {:error, :malformed_transaction} ->
+        case ExRLP.decode(tx2_binary) do
+          # first RLP item of the transaction specifies the tx type as `FeeTokenClaim` - can't test further
+          [<<3>> | _] -> true
+          # in all other cases the contract should revert
+          _ -> verify_both_error(tx2_binary, contract)
+        end
+
+      # in other cases of errors, we check whether both implementations reject the mutated transaction
+      {:error, _} ->
+        verify_both_error(tx2_binary, contract)
+    end
+  end
+
+  defp solidity_hash!(tx, contract) do
+    {:ok, solidity_hash} = solidity_hash(tx, contract)
+    solidity_hash
+  end
+
+  defp solidity_hash(%{} = tx, contract), do: tx |> Transaction.raw_txbytes() |> solidity_hash(contract)
+
+  defp solidity_hash(encoded_tx, contract) when is_binary(encoded_tx),
+    do: Eth.call_contract(contract, "hashTx(address,bytes)", [contract, encoded_tx], [{:bytes, 32}])
+
+  defp elixir_hash(%{} = tx), do: OMG.TypedDataHash.hash_struct(tx)
+  defp elixir_hash(encoded_tx), do: encoded_tx |> Transaction.decode!() |> elixir_hash()
+
+  defp assert_contract_reverted(result) do
+    Application.fetch_env!(:omg_eth, :eth_node)
+    |> case do
+      :ganache ->
```

Comment:
Is this needed?

Reply:
ah good point. Not sure. Ganache is friendlier with revert reasons, so this might be useful. 

Deciding if the reason is as expected has been dropped (we don't care about the reason) in the actual tests (we're using `:geth` with snapshotted contracts). But in case someone would like to run the suite against `ganache` for any reason, this is a useful clause. 

Other than that, we are still keeping support for `ganache` despite no active workflows use it. I used to use it quite a lot no more than ~2 months ago, it does come in handy.

I'd keep it, just like the rest of `ganache` support, for the time being. These things are confusing to figure out and we do switch clients used rather liberally. WDYT?